8.10.0 Release notes (#3781)
elastic/security-docs: docs/release-notes/8.10.asciidoc

[[release-notes-header-8.10.0]]
== 8.10

[discrete]
[[release-notes-8.10.0]]
=== 8.10.0

[discrete]
[[breaking-changes-8.10.0]]
==== Breaking changes
There are no breaking changes in 8.10.0.

[discrete]
[[features-8.10.0]]

> ==== New features

Author: Leaving this open because still need a PR link and blog link from @PhilippeOberti and @paulewing.
 
```suggestion
==== New features
* Introduces a redesigned alert details flyout that enhances your investigative flows ({pull}TBD[#TBD]).
```

Author: Will be linking to doc PRs in lieu of kibana PRs here.

```suggestion
==== New features
* Introduces a redesigned alert details flyout that enhances your investigative flows (https://github.com/elastic/security-docs/pull/3816)[#3816], (https://github.com/elastic/security-docs/pull/3854)[#3854).
```

> * Adds the MITRE ATT&CK® coverage page, which shows the MITRE ATT&CK® adversary tactics and techniques are covered by your installed and enabled detection rules ({pull}161556[#161556], {pull}163498[#163498], {pull}164613[#164613], {pull}164986[#164986]).

Reviewer: Nit: maybe a typo

```suggestion
* Adds the MITRE ATT&CK® coverage page, which shows how the MITRE ATT&CK® adversary tactics and techniques are covered by your installed and enabled detection rules ({pull}161556[#161556], {pull}163498[#163498], {pull}164613[#164613], {pull}164986[#164986]).
```

> * Adds the MITRE ATT&CK® coverage page, which shows how the MITRE ATT&CK® adversary tactics and techniques are covered by your installed and enabled detection rules ({pull}161556[#161556], {pull}163498[#163498], {pull}164613[#164613], {pull}164986[#164986]).

Reviewer 2: ```suggestion
* Adds the MITRE ATT&CK® coverage page, which shows how well your active detection rules protect against adversary tactics and techniques ({pull}161556[#161556], {pull}163498[#163498], {pull}164613[#164613], {pull}164986[#164986]).
```

* Adds a component under the Elastic AI Assistant header that allows you to select a different connector ({pull}163666[#163666]).
* Adds a flyout that allows you to examine rule details when installing or updating a prebuilt rule ({pull}163304[#163304]).
* Adds the ability to specify custom highlighted fields for an alert ({pull}163235[#163235]).

> * Adds role-based access control for the Elastic AI Assistant ({pull}163031[#163031]).

Reviewer 3: This isn't mandatory, but might make logical sense to group similar features together. So in this case, I'd put this RBAC for AI Assistant beneath the other AI feature on line 17. Same idea with rules, alerts, etc. Just a thought!

Author: Thanks, @jmikell821! Resolved in [2d9a7c0](https://github.com/elastic/security-docs/pull/3781/commits/2d9a7c078836966792b465741d0d0352a02f074b).

> * Adds role-based access control privileges for the Elastic AI Assistant ({pull}163031[#163031]).

Reviewer 2: ```suggestion
* Adds role-based access control for the Elastic AI Assistant ({pull}163031[#163031]).
```

* Adds the **Reputation service** option to the malicious behavior protection setting on the Elastic Defend integration policy ({pull}161617[#161617]).

[discrete]
[[enhancements-8.10.0]]
==== Enhancements
* Modifies the {elastic-sec} main menu by adding the Rules main page and landing page, and shifting the order of the Cases and Explore pages ({pull}165061[#165061], {pull}163102[#163102], {pull}161667[#161667]).

> * Adds the `resource.id` column to top ten vulnerability resources ({pull}162668[#162668]).

Reviewer 2: ```suggestion
* Adds a `resource.id` column to the "Top 10 vulnerable resources" table on the Cloud Native Vulnerability Management dashboard ({pull}162668[#162668]).
```

> * Findings tab redirection logic ({pull}162289[#162289]).

Reviewer 2: ```suggestion
* Shows the most relevant tab when you open the Findings page, instead of always showing the Misconfigurations tab ({pull}162289[#162289]).
```

> * Adds the **Building block** label to the rule details page to show whether a rule was a building block rule ({pull}162233[#162233]).

Reviewer 2: ```suggestion
* Adds the **Building block** label to the rule details page for building block rules ({pull}162233[#162233]).
```

* Removes a filter that restricted the fields user could choose from when creating an Endpoint exception or event filter ({pull}162193[#162193]).

> * A confirmation message displays on the Rules page rule details page when users delete rules ({pull}162477[#162477]). 

Reviewer: ```suggestion
* A confirmation message displays on the Rules page and rule details page when users delete rules ({pull}162477[#162477]). 
```

> * Removes a filter that restricted the fields user could choose from when creating an Endpoint exception or event filter ({pull}162193[#162193]).

Reviewer 2: ```suggestion
* Removes a filter that restricted the fields you could choose from when creating an Endpoint exception or event filter ({pull}162193[#162193]).
```

[discrete]
[[bug-fixes-8.10.0]]
==== Bug fixes

> * Fixes a UI bug that caused the rule preview to break when you closed it.  

Author: ```suggestion
* Fixes a UI bug that caused the rule preview to break when you closed it ({pull}164973[#164973]).  
```

> * A confirmation message displays on the Rules page and rule details page when users delete rules ({pull}162477[#162477]). 

Reviewer 2: ```suggestion
* Shows a confirmation message on the Rules page and rule details page when you delete rules ({pull}162477[#162477]). 
```

* Fixes a bug that stopped pre-configured connectors from working with the Elastic AI Assistant ({pull}164900[#164900]).

> * Adds the new Elastic AI Assistant logo and global header menu item ({pull}164763[#164763]).

Reviewer 2: Since it adds the header menu item I think this should be under enhancements

* Ensures that users see the appropriate message in the Elastic AI Assistant UI if they don't have the necessary connector and action privileges ({pull}164382[#164382]).

> * Stops threshold rules from displaying error message that conceal shard failure messages ({pull}164231[#164231]).

Reviewer 2: ```suggestion
* Prevents threshold rule error messages from concealing shard failure messages ({pull}164231[#164231]).
```

[[release-notes-header-8.10.0]]
== 8.10

[discrete]
[[release-notes-8.10.0]]
=== 8.10.0

[discrete]
[[breaking-changes-8.10.0]]
==== Breaking changes
There are no breaking changes in 8.10.0.

[discrete]
[[features-8.10.0]]
==== New features
* Adds a component under the Elastic AI Assistant header that allows you to select a different connector ({pull}163666[#163666]).
* Adds a flyout that allows you to examine rule details when installing or updating a prebuilt rule ({pull}163304[#163304]).
* Adds the ability to specify custom highlighted fields for an alert ({pull}163235[#163235]).
* Adds role-based access control privileges for the Elastic AI Assistant ({pull}163031[#163031]).

> * Adds the **Reputation service** option to the malicious behavior protection setting on the Elastic Defend integration policy ({pull}161617[#161617]).

Reviewer: We're missing a mention of the MITRE ATT&CK coverage page. It was implemented in the following PRs which were incorrectly labeled as `release_note:skip`:

- https://github.com/elastic/kibana/pull/161556
- https://github.com/elastic/kibana/pull/163498
- https://github.com/elastic/kibana/pull/164613
- https://github.com/elastic/kibana/pull/164986

Fixed all 4 to be `release_note:feature`.

Author: Thanks, @banderror! What should the summary for the new MITRE ATT&CK coverage page be?

Author: Thanks for catching this and updating the PR labels, @banderror! I added a summary for this PR in [7b5459e](https://github.com/elastic/security-docs/pull/3781/commits/7b5459ef15a2156cb75fe23ada95d5cd2e50f255).

[discrete]
[[enhancements-8.10.0]]

> ==== Enhancements

Reviewer: Here's another PR that I think was incorrectly labeled as `skip` and is rather a small UX enhancement: https://github.com/elastic/kibana/pull/162477

Author: Thanks for catching this and updating the PR labels, @banderror! I added a summary for this PR in [7b5459e](https://github.com/elastic/security-docs/pull/3781/commits/7b5459ef15a2156cb75fe23ada95d5cd2e50f255).

[[release-notes-header-8.10.0]]
== 8.10

[discrete]
[[release-notes-8.10.0]]
=== 8.10.0

[discrete]
[[breaking-changes-8.10.0]]
==== Breaking changes
There are no breaking changes in 8.10.0.

[discrete]
[[features-8.10.0]]
==== New features
* Adds a component under the Elastic AI Assistant header that allows you to select a different connector ({pull}163666[#163666]).
* Adds a flyout that allows you to examine rule details when installing or updating a prebuilt rule ({pull}163304[#163304]).
* Adds the ability to specify custom highlighted fields for an alert ({pull}163235[#163235]).
* Adds role-based access control privileges for the Elastic AI Assistant ({pull}163031[#163031]).
* Adds the **Reputation service** option to the malicious behavior protection setting on the Elastic Defend integration policy ({pull}161617[#161617]).

[discrete]
[[enhancements-8.10.0]]
==== Enhancements
* Modifies the {elastic-sec} main menu ({pull}159185[#159185]).

> * New sections with landing page ({pull}163102[#163102]).

Author: Removing as per @semd's feedback [here](https://github.com/elastic/security-docs/pull/3781/files#r1315517690).
 
```suggestion
* New sections with landing page ({pull}163102[#163102]).
```

> * New sections with landing page ({pull}163102[#163102]).

Author: Removing as per @semd's feedback [here](https://github.com/elastic/security-docs/pull/3781/files#r1315517690).
 
```suggestion
* New sections with landing page ({pull}163102[#163102]).
```

[[release-notes-header-8.10.0]]
== 8.10

[discrete]
[[release-notes-8.10.0]]
=== 8.10.0

[discrete]
[[breaking-changes-8.10.0]]
==== Breaking changes
There are no breaking changes in 8.10.0.

[discrete]
[[features-8.10.0]]
==== New features
* Adds a component under the Elastic AI Assistant header that allows you to select a different connector ({pull}163666[#163666]).
* Adds a flyout that allows you to examine rule details when installing or updating a prebuilt rule ({pull}163304[#163304]).
* Adds the ability to specify custom highlighted fields for an alert ({pull}163235[#163235]).
* Adds role-based access control privileges for the Elastic AI Assistant ({pull}163031[#163031]).

> * Unified IA Project Navigation ({pull}161667[#161667]).

Author: @natasha-moore-elastic I could use your help with the release note summary for this one :) It can be something brief like, "Updates the main menu in this way..."

> * Unified IA Project Navigation ({pull}161667[#161667]).

Author: Will incorporate this change if merge the summary for this into the summary written [here](https://github.com/elastic/security-docs/pull/3781/files#r1315161753).

```suggestion
```

* Adds the **Reputation service** option to the malicious behavior protection setting on the Elastic Defend integration policy ({pull}161617[#161617]).

[discrete]
[[enhancements-8.10.0]]
==== Enhancements

> * Move the Rules page up a level in the {elastic-sec} main menu  ({pull}165061[#165061]). 

Author: Will incorporate this change if merge the summary for this into the summary written [here](https://github.com/elastic/security-docs/pull/3781/files#r1315161753).

```suggestion
```

> * Modifies the {elastic-sec} main menu ({pull}159185[#159185]).

Author: @natasha-moore-elastic I could use your help with the release note summary for this one :) It can be something brief like, "Updates the main menu in this way..."

Reviewer 4: How about:
```suggestion
* Updates the style of the {elastic-sec} main menu, and adds a new Rules section. ({pull}159185[#159185]).
```

Author: Thanks for this, @natasha-moore-elastic! Since changes to the main nav are spread over a few PRs ([#165061](https://github.com/elastic/kibana/pull/165061), [#163102](https://github.com/elastic/kibana/pull/163102), and [#161667](https://github.com/elastic/kibana/pull/161667)), it might make sense to write a single release note summary for all three. Maybe something like this:

```suggestion
* Adds a Rules page and Rules landing page to the {elastic-sec} main menu ({pull}159185[#159185]).
```

@semd does this accurately convey the main nav changes in 8.10?

Reviewer 5: @nastasha-solomon Yes, and the `New sections with landing page` line should be removed, since Rules is the only new section and landing page for ESS (the rest are serverless only).

It would be an enhancement, no new feature has been introduced, only links relocations and UX improvements. fwiw, the visual changes are the following:

previous:
![previous](https://github.com/elastic/security-docs/assets/17747913/40aa2066-d390-4587-b08e-86cddc7e5500)

new:
![new](https://github.com/elastic/security-docs/assets/17747913/7eb5f7ce-5b1e-439d-91d4-be171ac1afe5)

Author: Thanks for the screenshots, @semd! It looks like the Cases and Explore pages were re-ordered as well. I can mention this in the summary if necessary:

```suggestion
* Modifies the {elastic-sec} main menu by adding the Rules main page and landing page, and shifting the order of the Cases and Explore pages ({pull}165061[#165061], {pull}163102[#163102], {pull}161667[#161667]).
```

* New sections with landing page ({pull}163102[#163102]).
* Adds the `resource.id` column to top ten vulnerability resources ({pull}162668[#162668]).

> * Findings tab redirection logic ({pull}162289[#162289]).

Author: @benironside can you provide a summary for this PR and PR #162668 as well? I wasn't sure what to write for either.

[[release-notes-header-8.10.0]]
== 8.10

[discrete]
[[release-notes-8.10.0]]
=== 8.10.0

[discrete]
[[known-issue-8.10.0]]
==== Known issues

* The alert prevalence feature behaves inconsistently and may not produce accurate results.

[discrete]
[[breaking-changes-8.10.0]]
==== Breaking changes
There are no breaking changes in 8.10.0.

[discrete]
[[features-8.10.0]]
==== New features

> * Introduces a redesigned alert details flyout that enhances your investigative flows (https://github.com/elastic/security-docs/pull/3816)[#3816],(https://github.com/elastic/security-docs/pull/3854)[#3854).

Author: ```suggestion
* Introduces a redesigned alert details flyout that enhances your investigative flows (https://github.com/elastic/security-docs/pull/3816)[#3816],(https://github.com/elastic/security-docs/pull/3854)[#3854]).
```

> * Introduces a redesigned alert details flyout that enhances your investigative flows (https://github.com/elastic/security-docs/pull/3816)[#3816], (https://github.com/elastic/security-docs/pull/3854)[#3854).

Author: ```suggestion
* Introduces a redesigned alert details flyout that enhances your investigative flows (https://github.com/elastic/security-docs/pull/3816)[#3816],(https://github.com/elastic/security-docs/pull/3854)[#3854).
```
